src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -244,7 +244,7 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
     def length: Int = unsafeArray.length
     @throws[ArrayIndexOutOfBoundsException]
     def apply(i: Int): Byte = unsafeArray(i)
-    override def hashCode = MurmurHash3.byteArraySeqHash(unsafeArray)
+    override def hashCode = MurmurHash3.arraySeqHash(unsafeArray)
```

Comment:
- `arraySeqHash` is defined as `arrayHash(a, seqSeed)`
- `byteArraySeqHash` is `bytesHash(data, seqSeed)`

the scaladoc of `bytesHash` says "Compute the hash of a byte array. Faster than arrayHash, because it hashes 4 bytes at once". this sounds to me like it should compute the same result, only more efficiently. shouldn't the fix go there instead?

Reply:
I don't think we can use `byteArraySeqHash`. The doc comment is misleading:

```
  /** Compute the hash of a byte array. Faster than arrayHash, because
   *  it hashes 4 bytes at once.
```

It is not only faster but also incompatible with `arrayHash`. The reason why It is  faster is that it uses a different algorithm. `arrayHash` is already specialized so it is as good as it gets without changing the hash code. `byteArrayHash` makes use of the fact that you can mix in a 32-bit value in each step by hashing 4 bytes per step.

Reply:
OK, thanks. As far as I can see, `bytesHash` is unused. Myabe it's better to remove it? It could be used for hashes that are only used internally, but we don't seem to have a use case. Otherwise we should at least improve the docs.